public/index.html

```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <link rel="stylesheet" href="assets/css/style.css">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.2/css/all.min.css">
+    <title>APIs</title>
+</head>
+
+<body>
+    <div class="categories-section">
+        <div class="heading-categories">
+            <h2 class='heading'>Categories:</h2>
+            <input type="text" id="searchBar" class='search-bar' placeholder="Search Category" maxlength="30"></input>
+            <button type="button" id="btnClear" class="btn-clear">Clear</button>
+        </div>
+        <div class='categories' id='categories'></div>
+    </div>
+
+    <div class="search-section" id="chosenCategorySearch">
+        <h3 class="heading">Selected:</h3>
+        <div id="displaySelected">
+            <!-- <div class="chosen-category-div">
+                <p id='chosenCategory' class='chosen-category'>asdadassssss</p>
+                <button type="button" class='chosen-category-btn'>X</button>
+            </div> -->
+        </div>
+    </div>
+
+    <div class="links-section">
+        <div class="heading-categories">
+            <h2 class='heading'>Links:</h2>
+            <input type="text" id="searchBarLinks" class='search-bar' placeholder="Search Links" maxlength="30"></input>
+            <button type="button" id="btnClearLinks" class="btn-clear">Clear</button>
+        </div>
+        <div class="one-category-box" id="oneCategoryDisplayBox"></div>
+    </div>
+<script type="text/javascript" src="main.bundle.js"></script></body>
+
```

Comment:
don't commit public/main bundle/map and everything build based, your mistake was for firebase to use public.
usually you could use dist/build for firebase and add it dist/build into git ignore.